packages/terra-form-select/src/native-select/NativeSelect.jsx

```diff
@@ -137,12 +137,12 @@ const createOptions = options => {
       currentOptGroupKeys.push(optGroupKey);
 
       return (
-        <optgroup key={optGroupKey} label={current.display}>
+        <optgroup disabled={!!current.disabled} key={optGroupKey} label={current.display}>
```

Comment:
We need to selectively add the disabled attribute as it's value isn't considered, it's a simple presence check for the browser. `disabled={!!current.disabled}` will result in either the attribute `disabled="true"` or `disabled="false"` being displayed on the DOM. The browser treats`disabled="false"` displays as disabled.

Reply:
Reverted b36953e